tmail-backend/apps/memory/src/main/java/com/linagora/tmail/james/app/MemoryServer.java

```diff
@@ -225,9 +234,36 @@ public static GuiceJamesServer createServer(MemoryConfiguration configuration) {
             .combineWith(chooseLinagoraServiceDiscovery(configuration.linagoraServicesDiscoveryModuleChooserConfiguration()))
             .combineWith(choosePop3ServerModule(configuration))
             .combineWith(chooseDropListsModule(configuration))
+            .combineWith(extentionModules(configuration))
             .overrideWith(chooseOpenPaas(configuration.openPaasModuleChooserConfiguration()))
             .overrideWith(chooseMailbox(configuration.mailboxConfiguration()))
-            .overrideWith(chooseJmapModule(configuration));
+            .overrideWith(chooseJmapModule(configuration))
+            .overrideWith(binder -> binder.bind(GuiceLoader.class).to(NoopGuiceLoader.class).in(Singleton.class));
+    }
+
+    private static Module extentionModules(MemoryConfiguration configuration) {
+        Injector injector = Guice.createInjector();
+        File jarFile = new File("src/main/extensions-jars/tmail-ai-bot-jar-with-dependencies.jar");
+        try {
+            URLClassLoader classLoader = new URLClassLoader(new URL[]{jarFile.toURI().toURL()}, MemoryServer.class.getClassLoader());
```

Comment:
Why?

 - 1. we can not hard code all the jars we will work with...
 - 2. we said the urlclassloader is likely not needed